rfcs/0000-single-cluster-dnspolicy.md

```diff
@@ -0,0 +1,286 @@
+As the `simple` strategy is set in the DNSPolicy a DNSRecord resource with the following contents will be created:
+
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSRecord
+metadata:
+  name: prod-web-api
+  namespace: my-gateways
+spec:
+  providerRef:
+    name: my-route53-credentials
+    namespace: my-gateways
+  endpoints:
+    - dnsName: myapp.mn.hcpapps.net
+      recordTTL: 60
+      recordType: A
+      targets:
+        - 172.31.200.0
+```
+
+The `providerRef` is included in the DNSRecord to allow the MGC dns record controller to load the appropriate provider during reconciliation and create the DNS records in the dns provider service e.g. route 53.
+
+**Example 2.** DNSPolicy using `simple` strategy with external dns provider
+
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+metadata:
+  name: prod-web
+  namespace: my-gateways
+spec:
+  providerRef:
+    name: external-dns
+    kind: ExternalDNS
+  targetRef:
+    name: prod-web
+    group: gateway.networking.k8s.io
+    kind: Gateway
+  strategy: simple
+```
+
+In ths example if the DNSPolicy was attached to the same gateway described in example 1, the same DNSRecord would also be created but MGC would not reconcile it. In this scenario it is expected that an external controller is being used to manage the reconciliation of the DNSRecord resources such as [external-dns](https://github.com/kubernetes-sigs/external-dns).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## API Updates
+
+DNSPolicy:
+
+* - new providerRef field `spec.providerRef` 
+* - new strategy field `spec.strategy` 
+
+DNSRecord:
+
+- `spec.managedZone` replaced with `spec.providerRef`
+
+### DNSPolicy.spec.providerRef
+
+The `providerRef` field is mandatory and contains a reference to a resource that shows how DNSRecords will be reconciled.
+    * - `spec.providerRef.name` - name of the provider resource
+    * - `spec.providerRef.namespace` - namespace of provider resource
+    * - `spec.providerRef.kind` - kind of resource, can be anything but only `Secret` will be reconciled by MGC.
+
+A kind of type `Secret` will be managed by MGC, and a secret in the given namespace with the given name must exist. The expected contents of the secrets data is comparable to the `dnsProviderSecretRef` used by ManageZones. 
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mgc-aws-credentials
+  namespace: multi-cluster-gateways
+type: kuadrant.io/aws
+data:
+  AWS_ACCESS_KEY_ID: "foo"
+  AWS_SECRET_ACCESS_KEY: "bar"
+  CONFIG:
+    zoneIDFilter:
+      - Z04114632NOABXYWH93QUl
+```
+
+The `CONFIG` section of the secrets data will be added to allow provider specific configuration to be stored alongside the providers credentials and can be used during the instantiation of the provider client, and during any provider operations.
```

Comment:
I wonder do we want to call out that this filter will decide whether or not a DNSRecord can be reconciled into the DNSProvider. If no appropriate zone is found, then we will fail to reconcile the the dnsrecord. Should there be some form of 
```listenerStatus``` on the DNSPolicy added that reflects whether the DNSPolicy is ready for that listener?
Also thikn that we should probably reflect some form of listener status condition on the gateway

Reply:
We might want a section in here around how we want status to look in the DNSPolicy as part of this?